PortalFetch/crawler_sel.py

```diff
@@ -0,0 +1,240 @@
+            submit = ipt
+            break
+
+    if not submit:
+        print("No submit button found！")
+        return
+
+    submit.click()
+    time.sleep(2)  # Waiting for completion
+
+    # Locate and click the Submit button
+    ipts = driver.find_elements_by_tag_name("input")
+    submit = ""
+    for ipt in ipts:
+        if "submit".lower() == ipt.get_attribute("type") and \
+                "Advanced Search" == ipt.get_attribute("value"):
+            submit = ipt
+            break
+
+    if not submit:
+        print("Advanced Search button not found！")
+        return
+
+    submit.click()
+    # Wait for a while to make sure the page loads
+    waitUtilPageLoaded(driver)
+    # Go to the advanced options page and start filling in various search terms
+    fillAdvanceSearch(driver)
+    # Save searched courses
+    html = saveResult(driver)  # see method
+    courseList = []
+    get_contents(courseList, html)  #Extract course information from html
+    save_contents(courseList) #Save course information as a csv file
+
+
+# login
+def login_myportal():
+    driver = webdriver.Chrome()
+    # Open the browser (The true url is：https://myportal.fhda.edu/）
+    driver.get("https://myportal.fhda.edu/")
+    # input username
+    username = int(input("Please enter your user name:  "))
+    driver.find_element_by_id("j_username").send_keys(username)
+    # input password
+    password = input("Please enter your password:   ")
+    driver.find_element_by_id("j_password").send_keys(password)
+    driver.find_element_by_id(
+        "btn-eventId-proceed").click()  # Wait for the response from the next page and make sure the page is loaded!
+
+
+# Open course search page
+def openSeachPage(driver):
+    """Open search page."""
+    # Find the Apps menu and click
+    menus = driver.find_elements_by_class_name("list-group-item")
+    #print("menus: ", menus)
+    if not menus:
+        print("No left menu found！")
+        return
+    appMenu = []
+    for menu in menus:
+        txt = menu.text
+        if ("apps".lower() == txt.lower()): #  No left menu found
+            appMenu = menu
+            break
+    if not appMenu:
+        print("Apps menu not found in left menu！")
+        return
+    time.sleep(2)  # Wait for the next page to come over
+    appMenu.click()  # Open the menu
+    # Find in App List（Look Up Classes）button
+    clss = driver.find_elements_by_class_name("myapps-item")
+    if not clss:
+        print("No app list found！")
+        return
+    classes = []
+    for cls in clss:
+        txt = cls.find_element_by_class_name("myapps-item-label").text
+        txt = str.strip(txt)
+        #print("text:  ",  txt)
+        if ("Look Up Classes".lower() == txt.lower()):
+            classes = cls
+            break
+    if not classes:
+        print("No Look Up Classes feature found in the app list!")
+        return
```

Comment:
For Cognitive Complexity, could write a separate method for this to find 'Look Up Classes' feature.